Integrate DT method for defining the PROTECTEDMEM region (#28184)
nrfconnect/sdk-nrf: samples/ironside_se/protectedmem_periphconf/README.rst

1. |connect_terminal|
#. Reset the development kit.

> The primary firmware uses :c:func:`flash_write` to write a test pattern to the last write block of ``protectedmem_partition``, then reboots after 1 second.
> The secondary firmware boots, logs the boot reason, uses :c:func:`flash_erase` to erase the corrupted region, and reboots after 1 second.
> On the subsequent boot, |ISE| finds the integrity check passing and launches the primary firmware again.

Reviewer: ```suggestion
```

I'd remove this part as it is a duplicate of the content from the previous chapter

Author: removed and added a bit more expected output, also mentioning that the sample continues executing in a loop

1. |connect_terminal|
#. Reset the development kit.

> Expected output from the primary firmware:

Reviewer: ```suggestion
The following is the expected output from the primary firmware:
```

.. code-block:: console

   Writing test pattern to protected memory...
   Rebooting after 1s. Secondary firmware should boot if protection is working.

> Expected output from the secondary firmware:

Reviewer: ```suggestion
The following is the expected output from the secondary firmware:
```

.. code-block:: console

   Secondary: Secondary firmware booted
   Secondary: Booted because of PROTECTEDMEM failure
   Secondary: Reverting test pattern write to protected memory...
   Secondary: Rebooting after 1s. Primary firmware should boot again since the corruption was fixed.

> The sample continues alternating between primary and secondary firmware indefinitely.

Reviewer: ```suggestion
The sample alternates indefinitely between the primary and secondary firmware.
```

Configuration
*************

The sample uses the following key configurations:

Shared Memory Map
  The ``dts/nrf54h20dk_nrf54h20_memory_map.dtsi`` file defines the partition layout for the board.
  It relocates ``periphconf_partition`` to be placed right after ``cpuapp_boot_partition`` at offset 0x40000, and defines a ``protectedmem_partition`` subpartition grouping both.
  A matching ``secondary_protectedmem_partition`` covers the secondary firmware's boot and periphconf partitions.
  Both ``app.overlay`` and ``secondary/app.overlay`` include this file, and ``sysbuild/uicr.overlay`` applies it to the UICR image so all images see the same partition layout.

Kconfig Configuration

>   The ``sysbuild/uicr.conf`` file enables PROTECTEDMEM (``CONFIG_GEN_UICR_PROTECTEDMEM=y``), secondary firmware (``CONFIG_GEN_UICR_SECONDARY=y``), and SECONDARY.PROTECTEDMEM (``CONFIG_GEN_UICR_SECONDARY_PROTECTEDMEM=y``).

Reviewer 2: use doc features e.g.:
```
:kconfig:option:`CONFIG_...`
:file:`...`
:zephyr_file:`...`
```

>   The PROTECTEDMEM size is derived automatically from the ``protectedmem_partition`` device tree node.

Reviewer: ```suggestion
  The PROTECTEDMEM size is derived automatically from the ``protectedmem_partition`` devicetree node.
```
